core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -155,9 +155,7 @@ public void onNext(HttpObject o) {
                     }
                     merged = headers;
                 } else {
-                    if (req.method() == HttpMethod.HEAD) {
-                        endOfStream = true;
```

Comment:
GET and HEAD request can have a body, although it's not used often, so this fix is not correct. We should perhaps touch the WebFlux side.

Reply:
Ah, never mind.

Reply:
Oh I couldn't get the relation between 
> GET and HEAD request can have a body 

and this fix...
Could you explain about this a little bit more?

I thought whether a HEAD request has a body or not does not matter 
because it's about handling response for HEAD - cancel subscription for response when response publishes body.

Reply:
No worries. My bad 😅